### PR TITLE
Add integration tests to check the SYSTEM scope behvaiour

### PR DESCRIPTION
**Pull request by sahandilshan**
### Purpose
* $subject
* Add integration tests to test the behavior of the SYSTEM scope

### Related PR(s)
* https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2004

*** Note
Merge after https://github.com/wso2/product-is/pull/15447

### CODE REVIEW

**Comment by senthalan** [comment]
Can we include a test where a non-privileged tenant user tries the flow as well?
And better if we can use password grant/auth code other than client credentials for this test. (better if we can keep cc and add other grant)

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/SystemScopePermissionValidationTestCase.java [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+    public void testIntrospectionEndpoint() throws Exception {
+
+        String scope = getScopesFromIntrospectionResponse();
+        doTheScopeValidationBasedOnTheTestUserMode(scope);
+    }
+
+    @Test(groups = "wso2.is", description = "Test introspection endpoint", dependsOnMethods = "testIntrospectionEndpoint")
+    public void getTokenAndValidate() throws Exception {
+
+        client = HttpClientBuilder.create().disableRedirectHandling().build();
```
Comment (indeewari) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/SystemScopePermissionValidationTestCase.java:173]: Is this protected by the finally block?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/SystemScopePermissionValidationTestCase.java [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+        org.json.simple.JSONObject introspectionResponse =
+                introspectTokenWithTenant(client, accessToken, introspectionUrl, username, userPassword);
+        Assert.assertTrue(introspectionResponse.containsKey("scope"));
+        return introspectionResponse.get("scope").toString();
+    }
+
+    private void doTheScopeValidationBasedOnTheTestUserMode(String scope) {
+
+        if (testUserMode == TestUserMode.SUPER_TENANT_ADMIN) {
+            Assert.assertTrue(scope.contains("internal_server_admin"), "Scope should contain " +
```
Comment (indeewari) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/SystemScopePermissionValidationTestCase.java:210]: Don't we have enums/constants for these "internal_server_admin" etc ?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/SystemScopePermissionValidationTestCase.java [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+        consumerSecret = appDto.getOauthConsumerSecret();
+        Assert.assertNotNull(consumerSecret, "Application creation failed.");
+    }
+
+    @Test(groups = "wso2.is", description = "Send authorize user request and get access token", dependsOnMethods = "testRegisterApplication")
+    public void testGetAccessToken() throws Exception {
+
+        List<NameValuePair> urlParameters = new ArrayList<>();urlParameters.add(new BasicNameValuePair("grantType",
+                OAuth2Constant.OAUTH2_GRANT_TYPE_CLIENT_CREDENTIALS));
+        urlParameters.add(new BasicNameValuePair("consumerKey", consumerKey));
```
Comment (indeewari) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/SystemScopePermissionValidationTestCase.java:136]: Don't we have enums/constants for these "consumerKey" etc?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/SystemScopePermissionValidationTestCase.java (hunk 1 above)
Comment (sahandilshan) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/SystemScopePermissionValidationTestCase.java:173]: No, I'll move this to the try block as well

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/SystemScopePermissionValidationTestCase.java (hunk 3 above)
Comment (sahandilshan) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/SystemScopePermissionValidationTestCase.java:136]: We have a constant for `consumerKey`, but not for other attributes like `consumerSecret`, `accessEndpoint` etc. Also, most of the tests have straightaway used these keys(without using any constant or enums) when they send similar types of requests.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/SystemScopePermissionValidationTestCase.java (hunk 2 above)
Comment (sahandilshan) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/SystemScopePermissionValidationTestCase.java:210]: Could not find any constants for these as well

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/SystemScopePermissionValidationTestCase.java (hunk 1 above)
Comment (sahandilshan) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/SystemScopePermissionValidationTestCase.java]: Fixed with https://github.com/wso2/product-is/pull/15437/commits/1a78966577c03d26c950091c0c927c6bdc183a54

**Review by senthalan — APPROVED** [review]

**Pull request merged**